src/main/java/org/ays/common/util/validation/EmailAddressValidator.java

```diff
@@ -39,6 +41,41 @@ public boolean isValid(String email, ConstraintValidatorContext constraintValida
         if (!StringUtils.hasText(email)) {
             return true;
         }
-        return email.matches(EMAIL_REGEX);
+
+        email = email.trim();
+        if (email.startsWith(" ") || email.endsWith(" ")) {
+            return this.buildViolation(constraintValidatorContext, "email must not start or end with whitespace");
+        }
+
+        if (email.chars().filter(ch -> ch == '@').count() != 1) {
+            return this.buildViolation(constraintValidatorContext, "email must contain exactly one '@' character");
+        }
+
+        if (email.matches(".*[()#\\[\\]\";,\\s].*")) {
+            return this.buildViolation(constraintValidatorContext, "email contains invalid special characters");
+        }
+
+
+        String[] parts = email.split("@", 2);
+        if (!Character.isLetterOrDigit(parts[0].charAt(0))) {
+            return this.buildViolation(constraintValidatorContext, "email local part must start with a letter or number");
+        }
+
+        String domainPart = parts[1];
+        if (domainPart.startsWith("-") || domainPart.endsWith("-")) {
+            return this.buildViolation(constraintValidatorContext, "domain must not start or end with a hyphen");
+        }
+
+        if (!email.matches(EMAIL_REGEX)) {
+            return this.buildViolation(constraintValidatorContext, "email is not in a valid format");
+        }
```

Comment:
```suggestion
            return this.buildViolation(constraintValidatorContext, "must not start or end with whitespace");
        }

        if (email.chars().filter(ch -> ch == '@').count() != 1) {
            return this.buildViolation(constraintValidatorContext, "must contain exactly one '@' character");
        }

        if (email.matches(".*[()#\\[\\]\";,\\s].*")) {
            return this.buildViolation(constraintValidatorContext, "contains invalid special characters");
        }


        String[] parts = email.split("@", 2);
        if (!Character.isLetterOrDigit(parts[0].charAt(0))) {
            return this.buildViolation(constraintValidatorContext, "local part must start with a letter or number");
        }

        String domainPart = parts[1];
        if (domainPart.startsWith("-") || domainPart.endsWith("-")) {
            return this.buildViolation(constraintValidatorContext, "domain must not start or end with a hyphen");
        }

        if (!email.matches(EMAIL_REGEX)) {
            return this.buildViolation(constraintValidatorContext, "must be valid format");
        }
```

Reply:
Buradaki mesajların geleceğine dair testleri göremedim. Her bir mesajı ayrı ayrı böldüğümüz için bence bu senaryolar için de testler eklemeliyiz.